resources/idlharness.js

```diff
@@ -2596,6 +2603,12 @@ IdlInterface.prototype.test_member_iterable = function(member)
 {
     subsetTestByKey(this.name, test, function()
     {
+        if (!this.should_have_interface_object()) {
+            return;
```

Comment:
Is this early return hit in any test? It looks a bit suspect to create and pass these tests without testing anything, and that we'd need to especially care about `this.should_have_interface_object()` for testing `iterator<T>` but not async iterable, for example.

In other words, does anything bad happen if these additions are not made, or is there possible Web IDL where we'd assert the wrong this without these changes, even if no spec currently does that?

Reply:
No, it doesn't.

I've added this check & assert because a) `test_member_stringifier()` has it and b) WebIDL spec does not forbid `iterable<>` on interfaces with [`[LegacyNoInterfaceObject]`](https://heycam.github.io/webidl/#LegacyNoInterfaceObject) extended attribute or callback interfaces.

However, I didn't find such `iterable<>` occurrence among WebKit's `*.idl` files (there is `stringifier` with `[NoInterfaceObject]` though), and both  [`[LegacyNoInterfaceObject]`](https://heycam.github.io/webidl/#LegacyNoInterfaceObject) and callback interfaces are legacy, so we are safe to remove it.